app/helpers/validation_helper.rb

```diff
@@ -0,0 +1,46 @@
+module ValidationHelper
+
+  class DataValidator
+    def get_form_errors(params, field_definitions, register_name)
+      field_to_sym = ->(field){ field.underscore.to_sym }
+
+      register_sym = field_to_sym.call(register_name)
+
+      register_model_class = Class.new() do
+        # This is required see: https://stackoverflow.com/a/14431888/238230
+        def self.model_name
+          ActiveModel::Name.new(self, nil, 'record')
+        end
+        include ActiveModel::Validations
+        field_definitions.each do |field|
+          field_sym = field_to_sym.call(field[:item]['field'])
+          attr_accessor field_sym
+          is_key = field_sym == register_sym
+
+          case field[:item]['datatype']
+            # Custom validators are defined in app/validators
+          when 'integer'
+            validates field_sym, numericality: { only_integer: true, message: "%{value} is not an integer" }, allow_blank: true
+          when  'curie'
+            validates field_sym, format:{ with: /\A\w+:\w+\z/, message: "%{value} is not a valid curie"}, allow_blank: true
+          when 'string'
+            validates field_sym, presence: { message: "Field %{attribute} is required"}, allow_blank: !is_key
```

Comment:
I am assuming the key will always be a string, so I only check strings to see if they should be mandatory fields. Let me know if this is not a correct assumption.